### ISSUE TITLE
"Competitive Example" (4): Tree LSTMs

### ISSUE THREAD

**Issue by neubig** [clab/dynet#1288]
Part of: https://github.com/clab/dynet/issues/1284

Currently we have an example of training TreeLSTMs here:
> https://github.com/neulab/dynet-benchmark/blob/master/dynet-py/treenn.py
> https://github.com/neulab/dynet-benchmark/blob/master/dynet-cpp/treenn.cc

This should be verified to be competitive with the reported results in the original paper and moved to the competitive examples directory.

**Comment by zhiyong1997** [comment]
Hi, all. I'd like to deal with this.

**Comment by zhiyong1997** [comment]
I have modified the original treenn code, and spent some time tuning hyper-parameters. My code can be accessed by https://github.com/zhiyong1997/dynet/tree/master/examples/treelstm.
I want to make a throughput comparison with TF fold. However, I find that dropout in dynet is very slow. I'm aware of #1154 , but it seems that I still meet with the problem. Without dropout, one epoch takes ~11 seconds, while with dropout, one epoch takes ~18 seconds. Is it a normal case?

**Comment by neubig** [comment]
Could you send a PR (labeled [WIP]) so we can test this easily and discuss there.

**Comment by xunzhang** [comment]
Resolved via https://github.com/clab/dynet/pull/1330. Thanks for the contribution @zhiyong1997 and @zhisbug !
